File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/utils/connectionData/InMemoryConnectionData.py
from javonet.utils.ConnectionType import ConnectionType
from javonet.utils.connectionData.IConnectionData import IConnectionData


class InMemoryConnectionData(IConnectionData):
    @property
    def connection_type(self):
        return ConnectionType.InMemory

    @property
    def hostname(self):
        return ""

    def serialize_connection_data(self):
        return [self.connection_type.value, 0, 0, 0, 0, 0, 0]

    def __eq__(self, other):
        return isinstance(other, InMemoryConnectionData)

    def __hash__(self):
        hash_code = 593727026
        hash_code = hash_code * -1521134295 + hash(self.connection_type)
        return hash_code

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Nodejs/utils/connectionData/InMemoryConnectionData.js
const ConnectionType = require('../../utils/ConnectionType');
const IConnectionData = require('./IConnectionData');

class InMemoryConnectionData extends IConnectionData {
    constructor() {
        super();
        this._connectionType = ConnectionType.IN_MEMORY;
        this._hostname = ""
    }

    get connectionType() {
        return this._connectionType;
    }

    get hostname() {
        return this._hostname;
    }

    serializeConnectionData() {
        return [this.connectionType, 0, 0, 0, 0, 0, 0];
    }

    equals(other) {
        return other instanceof InMemoryConnectionData;
    }
}

module.exports = InMemoryConnectionData;